clubs/serializers.py

```diff
@@ -587,37 +586,21 @@ class Meta:
         fields = ('id', 'name')
 
 
-class NoteSerializer(serializers.ModelSerializer):
+class NoteSerializer(ManyToManySaveMixin, serializers.ModelSerializer):
     creator = serializers.HiddenField(default=serializers.CurrentUserDefault())
     creating_club = serializers.SlugRelatedField(queryset=Club.objects.all(), slug_field='code')
     subject_club = serializers.SlugRelatedField(queryset=Club.objects.all(), slug_field='code')
     title = serializers.CharField(max_length=255, default='Note')
     content = serializers.CharField(required=False)
     note_tags = NoteTagSerializer(many=True, required=False)
 
-    def create(self, validated_data):
-        """
-        Manual create method because DRF does not support writable nested fields by default.
-        """
-        # assign tags to new club
-        note_tags = None
-
-        if 'note_tags' in validated_data:
-            note_tags = []
-            for note_tag in validated_data.pop('note_tags'):
-                (note_tag_object, _) = NoteTag.objects.get_or_create(**note_tag)
-                note_tags.append(note_tag_object)
-
-        # Note: it's important that all nested fields are POPPED from the validated data at this point. Otherwise,
-        # DRF will throw an error calling create on the superclass.
-        obj = super().create(validated_data)
-
-        if note_tags is not None:
-            obj.note_tags.set(note_tags)
-
-        return obj
-
     class Meta:
         model = Note
         fields = ('id', 'creator', 'creating_club', 'subject_club', 'title', 'content', 'note_tags',
                   'creating_club_permission', 'outside_club_permission')
+        save_related_fields = [
+            {
+                'field': 'note_tags',
```

Comment:
weird thought but would we want to migrate at some point to one `Tag` model that has a `ManyToManyField` that includes all models that can be associated with that `Tag`? Since I think we'll eventually have note tags, club tags, report tags, and event tags.

Reply:
This could work, but I don't think that the different tag groups would have much overlap. We could consider adding a `category` field to the tag model, but there are pros and cons when compared to using seperate models.

Combining the tag groups might make it harder for users to find applicable tags and might result in data entry errors (for example, a `Needs Response` tag might be applicable to a note, but would make no sense when applied to a club).